concepts/05 Widgets/File Manager/30 Validation/10 Permissions.md

```diff
@@ -0,0 +1,313 @@
+        });
+    });    
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-file-manager id="fileManager" 
+        [allowedFileExtensions]="allowedFileExtensions" >
+        <dxo-upload [maxFileSize]="1000000"></dxo-upload>
+        <dxo-permissions 
+            [create]="true"
+            [copy]="true"
+            [download]="true"
+            [move]="true"
+            [remove]="true"
+            [rename]="true"
+            [upload]="true">
+        </dxo-permissions>
+        <!-- ... -->
+    </dx-file-manager>
+
+    <!-- tab: app.component.ts -->
+    import { DxFileManagerModule } from 'devextreme-angular';
+
+    @Component({
+        selector: 'app-root',
+        templateUrl: './app.component.html',
+        styleUrls: ['./app.component.css']
+    })
+    export class AppComponent {
+        allowedFileExtensions: string[];
+        
+        constructor() {
+            this.allowedFileExtensions = [".txt", ".doc", ".png"];
+        }
+        //...
+    }
+
+    <!-- tab: app.module.ts -->
+    import { BrowserModule } from '@angular/platform-browser';
+    import { NgModule } from '@angular/core';
+    import { AppComponent } from './app.component';
+    import { DxFileManagerModule } from 'devextreme-angular';
+    
+    @NgModule({
+        declarations: [
+            AppComponent
+        ],
+        imports: [
+            BrowserModule,
+            DxFileManagerModule
+        ],
+        //...
+    })
+    export class AppModule { }    
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxFileManager :allowed-file-extensions="['.txt', '.doc', '.png']">   
+            <DxUpload :max-file-size="1000000" />
+            <DxPermissions
+                :create="true"
+                :copy="true"
+                :download="true"
+                :move="true"
+                :remove="true"
+                :rename="true"
+                :upload="true" />
+        </DxFileManager>
+    </template>
+    <script>
+        import 'devextreme/dist/css/dx.common.css';
+        import 'devextreme/dist/css/dx.light.css';    
+        
+        import { DxFileManager, DxPermissions } from 'devextreme-vue/file-manager';
+
+        export default {
+            components: {
+                DxFileManager,
+                DxPermissions
+            },
+            data() {
+                return {
+                    allowedFileExtensions,
+                    {/* ... */}
+                };
+            }            
+        };
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import FileManager, { 
+        Upload, Permissions 
+    } from 'devextreme-react/file-manager';
+    
+    class App extends React.Component {
+        render() {
+            return (
+                <FileManager allowedFileExtensions="['.txt', '.doc', '.png']" >
```

Comment:
It is better to not declare arrays, objects, functions in the markup to avoid unnecessary re-rendering.

```suggestion
                <FileManager allowedFileExtensions={allowedFileExtensions}>
```